stripe-servant/src/Stripe/Api.hs

```diff
@@ -40,10 +42,20 @@ type PriceApi
   :<|> StripeAuth :> Capture ":product_id" PriceId :> Get '[JSON] Price
   :<|> StripeAuth :> QueryParam "lookup_keys[]" T.Text :> Get '[JSON] (StripeList Price)
 
-type SubscriptionApi
-  = StripeAuth :> ReqBody '[FormUrlEncoded] SubscriptionCreate :> Post '[JSON] Subscription
-  :<|> StripeAuth :> Capture ":subscription_id" SubscriptionId :> Get '[JSON] Subscription
-  :<|> StripeAuth :> QueryParam "customer" CustomerId :> Get '[JSON] (StripeList Subscription)
+type InvoiceApi =
+  StripeAuth :> Capture ":invoice_id" InvoiceId :> Get '[JSON] Invoice
+    :<|> StripeAuth :> QueryParam "customer" CustomerId :> QueryParam "expand[]" T.Text :> Get '[JSON] (StripeList Invoice)
+
+type PaymentMethodApi =
+  StripeAuth :> QueryParam "customer" CustomerId :> QueryParam "type" T.Text :> Get '[JSON] (StripeList PaymentMethod)
+    :<|> StripeAuth :> Capture ":payment_method_id" PaymentMethodId :> "attach" :> QueryParam "customer" CustomerId :> Post '[JSON] PaymentMethod
```

Comment:
I think something like this would be cleaner:

```suggestion
    :<|> StripeAuth :> Capture ":payment_method_id" PaymentMethodId :> "attach" :> ReqBody '[FormUrlEncoded] PaymentMethodAttach :> Post '[JSON] PaymentMethod
```

with 

```
data PaymentMethodAttach 
     = PaymentMethodAttach
     { pmaCustomerId :: CustomerId
     }
```